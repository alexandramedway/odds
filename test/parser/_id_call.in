do PI
do myvar
do print(40 + 2)
do print("fourty-two")
do myfunc(arg1, arg2)
do noargs()
do prec(myvar, 2 * (2 + 3))
