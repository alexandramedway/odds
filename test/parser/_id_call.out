[
    Do(Id(PI)) ;
    Do(Id(myvar)) ;
    Do(Call(Id(print), [Binop(Int_lit(40), Add, Int_lit(2))])) ;
    Do(Call(Id(print), [String_lit(fourty-two)])) ;
    Do(Call(Id(myfunc), [Id(arg1) ; Id(arg2)])) ;
    Do(Call(Id(noargs), [])) ;
    Do(
        Call(
            Id(prec),
            [
                Id(myvar) ;
                Binop(Int_lit(2), Mult, Binop(Int_lit(2), Add, Int_lit(3)))
            ]
        )
    )
]
